OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -120,6 +120,19 @@ public override bool Tick(Actor self)
 			// and resume the activity after reloading if AbortOnResupply is set to 'false'
 			if (rearmable != null && !useLastVisibleTarget && attackAircraft.Armaments.All(x => x.IsTraitPaused || !x.Weapon.IsValidAgainst(target, self.World, self)))
 			{
+				// HACK: The gameplay expectation for AbortOnResupply is to cancel all queued activities, but only
+				// if the attack was explicitly ordered (i.e. FlyAttack is a top-level activity).
+				// Implicit orders (i.e. FlyAttack is a child activity) should not be disrupted
+				var isCurrentActivity = self.CurrentActivity == this || (self.CurrentActivity.State == ActivityState.Done && self.CurrentActivity.NextActivity == this);
+				if (attackAircraft.Info.AbortOnResupply && NextActivity != null && isCurrentActivity)
+				{
+					// HACK: Calling NextActivity.Cancel() will trigger a bug where cancelling queued
+					// orders will allow them to tick without calling OnFirstRun.
+					// Cancelling this activity *before* queueing the RTB child will properly
+					// skip NextActivity but still allow RTB to run to normal completion.
+					Cancel(self);
+				}
+
 				QueueChild(new ReturnToBase(self));
```

Comment:
Queuing this as a top level activity in the branch above should fix the target line issue.